### PR TITLE
Merge 1.4 into master

### PR DESCRIPTION
**Pull request by skeleton**

### CODE REVIEW

File: src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php [hunk 1]
```diff
@@ -397,4 +397,24 @@ public function getOrderedAndSortedByTreeCategories()
...
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function persistAsNextSiblingOf(CategoryInterface $node, CategoryInterface $prevSibling)
+    {
+        parent::persistAsNextSiblingOf($node, $prevSibling);
+
+        $this->_em->flush($node);
```
Comment (grena) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php]: To rework on master?

File: src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php [hunk 2]
```diff
@@ -397,4 +397,24 @@ public function getOrderedAndSortedByTreeCategories()
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function persistAsFirstChildOf(CategoryInterface $node, CategoryInterface $parent)
+    {
+        parent::persistAsFirstChildOf($node, $parent);
+
+        $this->_em->flush($node);
```
Comment (grena) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php]: To rework on master?

File: src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php (hunk 1 above)
Comment (jjanvier) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php]: yes

File: src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php (hunk 2 above)
Comment (jjanvier) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php]: yes

File: features/Context/Page/Base/Form.php [hunk 3]
```diff
@@ -257,25 +257,22 @@ public function openAvailableAttributesMenu()
...
-                throw new \Exception(sprintf('Could not find available attribute "%s".', $attribute));
-            }
+        foreach ($attributes as $attributeLabel) {
+            $search->setValue($attributeLabel);
+            $list  = $this->getElement('Available attributes list');
+            $label = $this->spin(
+                function () use ($list, $attributeLabel) {
+                    return $list->find('css', sprintf('li label:contains("%s")', $attributeLabel));
+                },
+                20,
```
Comment (phaseinducer) [inline features/Context/Page/Base/Form.php]: Doesn't exist anymore on master (cf. https://github.com/akeneo/pim-community-dev/pull/3473)

File: features/Context/Page/Product/Edit.php [hunk 4]
```diff
@@ -212,18 +212,35 @@ public function switchCopySource($source)
...
-            )
-        );
+        $list = $this->getElement('Available attributes list');
+        if (!$list->isVisible()) {
+            $this->openAvailableAttributesMenu();
+        }
+
+        $options = $this->spin(function () use ($list) {
+            return $list->findAll('css', 'li');
+        }, 20, 'No attributes found in available attributes list');
```
Comment (phaseinducer) [inline features/Context/Page/Product/Edit.php]: same

File: README.md [hunk 5]
```diff
@@ -10,12 +10,10 @@ Practically, it means the Akeneo PIM bundles are present in the src/ directory.
 If you want to contribute to the Akeneo PIM (and we will be pleased if you do!), you can fork
 this repository and submit a pull request.
 
-| [Master][Master] | [1.4][1.4] | [1.3][1.3] | [1.2][1.2] |
-|:----------------:|:----------:|:----------:|:----------:|
-| [![Build status][Master image]][Master] | [![Build status][1.4 image]][1.4] | [![Build status][1.3 image]][1.3] | [![Build status][1.2 image]][1.2] |
+| [1.4][1.4] | [1.3][1.3] | [1.2][1.2] |
+|:----------:|:----------:|:----------:|
+| [![Build status][1.4 image]][1.4] | [![Build status][1.3 image]][1.3] | [![Build status][1.2 image]][1.2] |
```
Comment (phaseinducer) [inline README.md]: Why change that ?

File: src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php [hunk 6]
```diff
@@ -82,4 +125,37 @@ protected function copyMedia(array $media)
...
+     */
+    protected function writeProductToBuffer(array $product)
+    {
+        if (!$this->bufferFile) {
+            $this->bufferFile = tempnam(sys_get_temp_dir(), 'pim_products_buffer_');
+        }
+
+        file_put_contents($this->bufferFile, json_encode($product) . "\n", FILE_APPEND);
+
+        $this->headers = $this->getAllKeys([
```
Comment (skeleton) [inline src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php]: should not be changed in master ?

Comment (phaseinducer) [inline src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php]: I'm reworking it right now in a separated PR : https://github.com/akeneo/pim-community-dev/pull/3563

**Pull request merged**